eng/pipelines/templates/steps/merge-parent-if-typespec-next.yml

```diff
@@ -0,0 +1,24 @@
+steps:
+  - script: git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" merge -Xours origin/main
+    displayName: git merge origin/main
+    condition: |
+      and(
+        succeeded(),
+        eq(variables['Build.Repository.Name'], 'azure-rest-api-specs'),
+        or(
+          eq(variables['Build.SourceBranchName'], 'typespec-next'),
+          eq(variables['System.PullRequest.targetBranchName'], 'typespec-next')
+        )
+      )
+
+  - script: git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" merge -Xours origin/RPSaaSMaster
+    displayName: git merge origin/RPSaaSMaster
```

Comment:
This has the potential to fail with conflicts. I think it would probably better to add the typespec-next (public and private) branch into our automation that creates merge PRs (similar to what we do for RPSaaSMaster). That way if there are issues or conflicts they can be resolved in that PR and we keep the branch updated.

Reply:
We could add these branches to our mirroring pipeline.  However, one difference is we want to merge the whole repo, including the `specification` folder.

Reply:
I agree merging during the build has a lot of problems, so I will try moving this to the mirror pipeline.